OpenRA.Mods.Common/Lint/CheckTranslationReference.cs

```diff
@@ -52,14 +47,48 @@ void ILintPass.Run(Action<string> emitError, Action<string> emitWarning, ModData
 							if (referencedKeys.Contains(key))
 								continue;
 
-							if (!translation.HasMessage(key))
-								emitError($"{key} not present in {language} translation.");
-
+							testKey(key);
 							referencedKeys.Add(key);
 						}
 					}
 				}
 			}
+		}
+
+		void ILintMapPass.Run(Action<string> emitError, Action<string> emitWarning, ModData modData, Map map)
+		{
+			// TODO: Check all available languages.
+			var language = "en";
+			var modTranslation = new Translation(language, modData.Manifest.Translations, modData.DefaultFileSystem);
+
+			var mapTranslation = map.TranslationDefinitions == null ? null :
```

Comment:
We can/should return immediately if `map.TranslationDefinitions == null` to avoid redundant checks.

Reply:
makes sense